src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -184,6 +192,16 @@
                         getProjectFilesystem(), getBuildTarget(), "%s__swiftcompile.argsfile")))
           : Optional.empty();
 
+    this.vfsoverlayFilePath =
+      swiftBuckConfig.getUseVFSOverlay()
+        ? Optional.of(
+        getProjectFilesystem()
+          .getRootPath()
+          .resolve(
+            BuildTargetPaths.getScratchPath(
+              getProjectFilesystem(), getBuildTarget(), "%s__vfsoverlay.yaml")))
+        : Optional.empty();
```

Comment:
Looks like `vfsoverlayFilePath` is guaranteed to be `empty` if `getUseVFSOverlay` is turned off.

Reply:
Yeah, I didn't invent this. The `filelist.txt` and `swiftcompile.argsfile` (a few lines above) are constructed like this way.